### PR TITLE
Update from upstream

### PR DESCRIPTION
**Pull request by chmeliik**
1. Add helper scripts for updating from upstream
2. Update from upstream

Bring project up to date with https://github.com/nexB/scancode-toolkit/commit/a15174f

Manual changes after running update script:
- copy _unicode_text_lines helper function
- drop unwanted formatting changes
- drop nested import that we replace with our own import
- verify that no updates are needed for test_analysis.py

### CODE REVIEW

**Comment by chmeliik** [comment]
PTAL @akostadinov (just please check if I caught all the relevant changes for BUNDLED WITH)

**Review by akostadinov — APPROVED** [review]
Everything seems to be there, thank you!

**Comment by chmeliik** [comment]
PTAL team, the upstream update is pretty simple, you can but don't have to check the helper script for upstream updates as well :slightly_smiling_face:

File: README.md [hunk 1]
```diff
@@ -4,3 +4,20 @@ This repository contains Gemfile.lock parser vendored from [ScanCode
...
 make sure that RubyGems dependencies are parsed without executing arbitrary Ruby
 code.
+
+## Updating
+
+To update the repository based on upstream changes, run `hack/update-from-upstream.sh`.
+
+* for `gemfile_lock.py` and its tests, the script updates everything below the import
+  block
+* for the supporting files, the script updates the existing top-level definifions
```
Comment (ejegrova) [inline README.md]: typo: definitions

Comment (chmeliik) [reply README.md]: fixed :+1:

**Review by ejegrova — APPROVED** [review]

**Pull request merged**